Source/WebKit/UIProcess/Extensions/WebExtension.cpp

```diff
@@ -245,6 +247,61 @@ void WebExtension::populateWebAccessibleResourcesIfNeeded()
         parseWebAccessibleResourcesVersion2();
 }
 
+URL WebExtension::resourceFileURLForPath(const String& originalPath)
+{
+    ASSERT(originalPath);
+
+    String path = originalPath;
+
+    if (path.startsWith('/'))
+        path = path.substring(1);
+
+    if (!path.length() || m_resourceBaseURL.isNull())
+        return { };
+
+    auto resourceURL = FileSystem::realPath(URL(m_resourceBaseURL, path).fileSystemPath());
+
+    // Don't allow escaping the base URL with "../".
+    if (!resourceURL.startsWith(m_resourceBaseURL.string())) {
```

Comment:
You need to get the real path to `m_resourceBaseURL.filesystemPath()` for this to work.